courseraprogramming/utils.py

```diff
@@ -129,13 +129,9 @@ def docker_client(args):
                 timeout=args.timeout,
                 **kwargs_from_env())
     elif _platform == 'darwin':
-        # OS X - Assume boot2docker, and pull from that environment.
+        # OS X.
         kwargs = kwargs_from_env()
-        if len(kwargs) == 0:
-            logging.error('Could not correctly pull in docker environment. '
```

Comment:
Good question. When there's a client creation error, think the prescription for both the `docker-machine` and native docker versions is to make sure that docker is actually running on the host. Added an error message to that effect in latest version.